src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,30 @@
+package seedu.addressbook.data.person;
+
+public abstract class Contact {
+
+	public String value;
+	public boolean isPrivate;
+
+	@Override
+	public String toString() {
+		return value;
+	}
+
+	@Override
+	public boolean equals(Object other) {
+		return other == this // short circuit if same object
+				|| (other instanceof Address // instanceof handles nulls
+						&& this.value.equals(((Address) other).value)); // state
```

Comment:
Contacts may also be email or phone objects. How does this method handle objects of those classes?